src/huggingface_hub/hf_api.py

```diff
@@ -244,11 +244,18 @@ class LastCommitInfo(TypedDict, total=False):
     date: datetime
 
 
-class BlobLfsInfo(TypedDict, total=False):
+@dataclass
+class BlobLfsInfo(dict):
     size: int
     sha256: str
     pointer_size: int
 
+    def __init__(self, data: dict):  # hack to make BlobLfsInfo backward compatible
+        self.__dict__.update(data)
+        self.size = data["size"]
+        self.sha256 = data["sha256"]
+        self.pointer_size = data["pointer_size"]
+
```

Comment:
I realized changing to `def __init__(self, data: dict):` is really backward compatible. What we can do instead is to use the `__post_init__` method to update the internal dict once all attributes have been set. It is also cleaner this way as we don't have to manually redefine the attributes.

```suggestion
    def __post_init__(self):
        # hack to make BlobLfsInfo backward compatible
        self.update(asdict(self))

```

:warning: you must add `from dataclasses import asdict` at the top of the file